### PR TITLE
Use constructors to construct objects

### PR DESCRIPTION
**Pull request by tautschnig**
This is some drive-by cleanup resulting from reviewing #3067. Instead of
incrementally constructing objects do RAII.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] My contribution is formatted in line with CODING_STANDARD.md.
- [n/a] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [n/a] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [n/a] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp [hunk 1]
```diff
@@ -320,14 +321,11 @@ static void instrument_start_thread(
   codet tmp_a(ID_start_thread);
   tmp_a.set(ID_destination, lbl1);
   code_gotot tmp_b(lbl2);
-  code_labelt tmp_c(lbl1);
-  tmp_c.op0() = codet(ID_skip);
+  const code_labelt tmp_c(lbl1, codet(ID_skip));
```
Comment (kroening) [inline jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp:324]: Use code_skipt, while at it?

File: jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp [hunk 2]
```diff
@@ -368,8 +366,7 @@ static void instrument_end_thread(
   // A: codet(id=ID_end_thread)
   // B: codet(id=ID_label,label= __CPROVER_THREAD_EXIT_<ID>)
   codet tmp_a(ID_end_thread);
-  code_labelt tmp_b(lbl2);
-  tmp_b.op0() = codet(ID_skip);
+  const code_labelt tmp_b(lbl2, codet(ID_skip));
```
Comment (kroening) [inline jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp:369]: same here

File: jbmc/src/java_bytecode/java_static_initializers.cpp [hunk 3]
```diff
@@ -208,18 +208,16 @@ static void clinit_wrapper_do_recursive_calls(
     auto findit = symbol_table.symbols.find(base_init_routine);
     if(findit == symbol_table.symbols.end())
       continue;
-    code_function_callt call_base;
-    call_base.function() = findit->second.symbol_expr();
-    init_body.move_to_operands(call_base);
+    const code_function_callt call_base(findit->second.symbol_expr());
+    init_body.copy_to_operands(call_base);
   }
```
Comment (kroening) [inline jbmc/src/java_bytecode/java_static_initializers.cpp:213]: .add()

File: jbmc/src/java_bytecode/java_static_initializers.cpp [hunk 4]
```diff
@@ -208,18 +208,16 @@ static void clinit_wrapper_do_recursive_calls(
...
   const irep_idt &real_clinit_name = clinit_function_name(class_name);
   auto find_sym_it = symbol_table.symbols.find(real_clinit_name);
   if(find_sym_it != symbol_table.symbols.end())
   {
-    code_function_callt call_real_init;
-    call_real_init.function() = find_sym_it->second.symbol_expr();
-    init_body.move_to_operands(call_real_init);
+    const code_function_callt call_real_init(find_sym_it->second.symbol_expr());
+    init_body.copy_to_operands(call_real_init);
   }
```
Comment (kroening) [inline jbmc/src/java_bytecode/java_static_initializers.cpp:221]: .add()

Comment (tautschnig) [reply jbmc/src/java_bytecode/java_static_initializers.cpp:221]: @kroening  All done!

**Review by kroening — APPROVED** [review]

File: jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp [hunk 5]
```diff
@@ -122,11 +122,8 @@ static codet get_monitor_call(
     return code_skipt();
 
   // Otherwise we create a function call
-  code_function_callt call;
-  call.function() = symbol_exprt(symbol, it->second.type);
-  call.lhs().make_nil();
-  call.arguments().push_back(object);
-  return call;
+  return code_function_callt(
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp]: Add a constructor for a call with no LHS?

**Review by smowton — APPROVED** [review]
One simple improvement given the code-function-callt-with-dummy-first-argument pattern occurs a few times (though perhaps this would introduce an ambiguous first constructor?) Also as noted in the other PR, I'd prefer if move-to-operands -> add was not pessimised.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 11d34d3a63db9f65b8c31d5a4502c4e7d1e7c0d6).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86479380
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 0ee10a48f424cb91ea558ecf8a69c1e945aaeacf).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86484504

File: jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp (hunk 5 above)
Comment (tautschnig) [reply jbmc/src/java_bytecode/java_bytecode_concurrency_instrumentation.cpp]: Done - found quite a few occurrences of this!

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: dbc543971c196cf047bc989afa2e1ad9433d0cc1).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86678835

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**